design/src/rfcs/rfc0037_retry_classifier_customization.md

```diff
@@ -0,0 +1,240 @@
+# RFC: User-configurable retry classification
+
+> Status: RFC
+>
+> Applies to: client
+
+For a summarized list of proposed changes, see the [Changes Checklist](#changes-checklist) section.
+
+This RFC defines the user experience and implementation of user-configurable retry classification. Custom retry classifiers enable users to change what errors are retried while still allowing them to rely on defaults set by SDK authors when desired.
+## Terminology
+
+- **Smithy Service**: An HTTP service, whose API is modeled with the [Smithy IDL](https://www.smithy.io).
+- **Smithy Client**: An HTTP client generated by smithy-rs from a `.smithy` model file.
+- **AWS SDK**: A **smithy client** that's specifically configured to work with an AWS service.
+- **Operation**: A modeled interaction with a service, defining the proper input and expected output types, as well as important metadata related to request construction. "Sending" an operation implies sending one or more HTTP requests to **smithy service**, and then receiving an output or error in response.
+- **Orchestrator**: The client code which manages the request/response pipeline. The orchestrator is responsible for: 
+	- Constructing, serializing, and sending requests.
+	- Receiving, deserializing, and (optionally) retrying requests.
+	- Running interceptors *(not covered in this RFC)* and handling errors.
+- **Runtime Component**: A part of the orchestrator responsible for a specific function. Runtime components are *always* required and may depend on specific configuration. Examples include the endpoint resolver, retry strategy, and request signer.
```

Comment:
```suggestion
- **Runtime Component**: A part of the orchestrator responsible for a specific function. Runtime components are used by the orchestrator itself, may depend on specific configuration, and must not be changed by interceptors. Examples include the endpoint resolver, retry strategy, and request signer.
```
The `http_client`/`http_connector` and `sleep_impl` components are actually optional. The HTTP piece isn't used for presigned requests, and the async sleep isn't used in WASM targets.